python/app/work_area_button.py

```diff
@@ -0,0 +1,303 @@
+        """
+        Sets up the button for a given entity.
+        Typically implemented by subclasses.
+
+        :param entity_type: Entity type to set up button for
+        :param entity_id: Entity id to set up button for
+        """
+        self._entity_id = entity_id
+        self._entity_type = entity_type
+
+        if not self._bundle.get_setting("enable_context_switch"):
+            # context switch button not enabled
+            return
+
+        # figure out if this is the current project
+        context = self._bundle.context
+        context_entity = context.task or context.entity or context.project or None
+
+        self.setVisible(True)
+
+        if context_entity and context_entity["type"] == entity_type and context_entity["id"] == entity_id:
+            self._is_current = True
+            self.setPopupMode(QtGui.QToolButton.DelayedPopup)
+            self.setToolTip(
+                "This is your current work area.\n"
+                "The work you do will be associated with this item in Shotgun."
+            )
+
+        elif entity_type in self.NON_WORK_AREA_TYPES:
+            # don't show the ctx selector for some types
+            self.setVisible(False)
+
+        else:
+            self._is_current = False
+
+            if entity_type == "Task":
+                self._caption = "Set Work Area"
+                self.setToolTip("Click to set your work area to the current task.")
+
+            else:
+                self._caption = "Pick Work Area"
+                self.setToolTip("Click to select a task.")
+
+        # button cannot be clicked
+        self.setEnabled(not self._is_current)
+
+        # tell the style sheet to adjust
+        self.setProperty("is_current", self._is_current)
+        self.style().unpolish(self)
+        self.style().polish(self)
+
+        self._init_default_state()
+
+    def _init_default_state(self):
+        """
+        Sets up the default collapsed state of the button
+        """
+        self.setText("")
+        self.setToolButtonStyle(QtCore.Qt.ToolButtonIconOnly)
+        self.setMinimumSize(QtCore.QSize(self.WIDGET_WIDTH_COLLAPSED, self.WIDGET_HEIGHT))
+        self.setMaximumSize(QtCore.QSize(self.WIDGET_WIDTH_COLLAPSED, self.WIDGET_HEIGHT))
+        # tell the style sheet to adjust
+        self.setProperty("is_expanded", False)
+        self.style().unpolish(self)
+        self.style().polish(self)
+
+    def _on_click(self):
+        """
+        Executed when the button is clicked
+        """
+        self.change_work_area.emit(self._entity_type, self._entity_id)
+
+    def enterEvent(self, evt):
+        """
+        QT Mouse enter event
+        """
+        if not self._is_current:
+            # not the current work area. so expand the button
+            self.setText(self._caption)
+            self.setToolButtonStyle(QtCore.Qt.ToolButtonTextBesideIcon)
+            self.setMinimumSize(QtCore.QSize(self._width, self.WIDGET_HEIGHT))
```

Comment:
Curious if you tried to use stylesheets to handle the hover look? I think I know the answer given our chat the other day about stylesheets. :)